storm/src/main/storm/mesos/logviewer/LogViewerController.java

```diff
@@ -31,13 +31,19 @@
 
 public class LogViewerController {
   private static final Logger LOG = LoggerFactory.getLogger(LogViewerController.class);
+  private static final String STORM_LOG_DIR_CONF = "storm.log.dir";
   protected Process process;
   protected SocketUrlDetection urlDetector;
   protected Integer port;
+  protected String logdir;
 
   public LogViewerController(Map conf) {
     port = Optional.fromNullable((Number) conf.get(Config.LOGVIEWER_PORT)).or(8000).intValue();
     setUrlDetector(new SocketUrlDetection(port));
+    logdir = System.getenv("MESOS_SANDBOX") + "/logs";
+    if (conf.containsKey(STORM_LOG_DIR_CONF)) {
+      logdir = (String) conf.get(STORM_LOG_DIR_CONF);
+    }
```

Comment:
Along these lines do we need to change line 111 to:
```
Path logPath = Paths.get(System.getenv("MESOS_SANDBOX"), "/logs");
```  
Similarly change line 115 to:
```
File log = Paths.get(logDir, "/logviewer-startup.log").toFile();
```